src/shared/components/palette/case-link/write-case-link-field.html

```diff
@@ -1,8 +1,8 @@
-<div class="form-group" [ngClass]="{'form-group-error': !caseReferenceControl.valid && caseReferenceControl.dirty}">
+<div class="form-group" [ngClass]="{'form-group-error': !caseReferenceControl.valid && (caseReferenceControl.dirty || caseReferenceControl.touched)}">
   <label [for]="id()">
     <span class="form-label" *ngIf="caseField.label">{{caseField | ccdFieldLabel}}</span>
   </label>
   <span class="form-hint" *ngIf="caseField.hint_text">{{caseField.hint_text}}</span>
-  <span class="error-message" *ngIf="caseReferenceControl.errors && caseReferenceControl.dirty">{{caseReferenceControl.errors | ccdFirstError:caseField.label}}</span>
+  <span class="error-message" *ngIf="caseReferenceControl.errors && (caseReferenceControl.dirty || caseReferenceControl.touched)">{{caseReferenceControl.errors | ccdFirstError:caseField.label}}</span>
```

Comment:
```suggestion
extract the method to typescript:
public isControlInvalid(): boolean {
return caseReferenceControl.invalid && (caseReferenceControl.dirty || caseReferenceControl.touched);
}

Then replace it in the two places in html:
*ngIf="isControlInvalid"
and:
[ngClass]="{'form-group-error':isControlInvalid}"

This will avoid duplicated caculation in html, improve the html performance a bit.
Also it will be more concise in the code.
```

Reply:
@rayliangatos I don't think I agree with you on improving performance yes it might be more concise. Using an angular method will cause angular to evaluate the method multiple times. Try still in a console.log in a method you can observe the issue.

Reply:
it's nothing related to whether you put the logic in the typescript or html, that is the ngIf directive's behaviour, it is asyn and "watches" for the variable's value. Even now you put all these in hmtl, it is still listening to the control's status all the time, it's just you couldn't console log this in html.

Reply:
https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496